lib/src/args_wrapper.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:meta/meta.dart';
+
+abstract class Arg<T> {
+  String name;
+
+  final String help;
+  final String abbr;
+  final bool hide;
+
+  String valueHelp;
+
+  Arg._(
+    this.name, {
+    @required this.abbr,
+    @required this.help,
+    @required this.hide,
+  });
+}
+
+class Flag extends Arg<bool> {
+  final bool negatable;
+  final bool defaultsTo;
+
+  Flag(
+    String name, {
+    this.negatable,
+    String abbr,
+    String help,
+    bool hide,
+    this.defaultsTo,
+  }) : super._(name, abbr: abbr, help: help, hide: hide);
+}
+
+class MultiOption extends Arg<List<String>> {
+  final bool splitCommas;
+  List<String> allowed;
+  Map<String, String> allowedHelp;
+  final List<String> defaultsTo;
+  MultiOption(String name,
+      {this.splitCommas,
+      bool hide = false,
+      String help,
+      String abbr,
+      this.allowed,
+      this.allowedHelp,
+      this.defaultsTo})
+      : super._(name, abbr: abbr, help: help, hide: hide);
+}
+
+class Option extends Arg<String> {
+  final List<String> allowed;
+  final Map<String, String> allowedHelp;
+  final String defaultsTo;
+  Option(String name,
+      {String abbr,
+      String help,
+      this.allowed,
+      this.allowedHelp,
+      this.defaultsTo,
+      bool hide})
+      : super._(name, abbr: abbr, help: help, hide: hide);
+}
+
+extension ArgResultExt on ArgResults {
+  T get<T>(Arg<T> arg) => this[arg.name] as T;
```

Comment:
Ideally, you'd probably want to make this throw an Error of the `arg` instance wasn't registered with the associated `ArgParser`.

That requires adding and storing state on `ArgParser`.